Wrappers/Python/ccpi/optimisation/algorithms/ADMM_linearized.py

```diff
@@ -0,0 +1,280 @@
+from __future__ import unicode_literals
+
+from ccpi.optimisation.algorithms import Algorithm
+
+class ADMM_linearized(Algorithm):
+        
+    ''' 
+        Linearized Alternating Direction Method of Multipliers (ADMM)
+    
+        General form of ADMM : min_{x} f(x) + g(y), subject to Ax + By = b
+    
+        Case: A = Id, B = -K, b = 0   ==> min_x f(Kx) + g(x)  
+                
+        The quadratic term in the augmented Lagrangian is linearized for the x-update.
+                            
+        Main algorithmic difference is that in ADMM we compute two proximal subproblems, 
+        where in the PDHG a proximal and proximal conjugate.
+
+        Reference (Section 8) : https://link.springer.com/content/pdf/10.1007/s10107-018-1321-1.pdf                          
+
+            x^{k} = prox_{\tau f } (x^{k-1} - tau/sigma A^{T}(Ax^{k-1} - z^{k-1} + u^{k-1} )                
+            
+            z^{k} = prox_{\sigma g} (Ax^{k} + u^{k-1})
+            
+            u^{k} = u^{k-1} + Ax^{k} - z^{k}
+                
+    '''
+
+    def __init__(self, f, g, operator, \
+                       tau = None, sigma = 1., 
+                       x_init = None):
+        
+        '''Initialisation of the algorithm
+
+        :param operator: a Linear Operator
+        :param f: Convex function with "simple" proximal
+        :param g: Convex function with "simple" proximal 
+        :param sigma: Positive step size parameter 
+        :param tau: Positive step size parameter
+        :param x_init: Initial guess ( Default x_init = 0)'''        
+        
+        super(ADMM_linearized, self).__init__()
+        
+        self.set_up(f = f, g = g, operator = operator, tau = tau, sigma = sigma, x_init = x_init)        
+                    
+    def set_up(self, f, g, operator, tau = None, sigma = 1., x_init = None):
+
+        print("{} setting up".format(self.__class__.__name__, ))
+        
+        if sigma is None and tau is None:
+            raise ValueError('Need tau <= sigma / ||K||^2')
+
+        self.f = f
+        self.g = g
+        self.operator = operator
+
+        self.tau = tau
+        self.sigma = sigma
+
+        if self.tau is None:
+            normK = self.operator.norm()
+            self.tau = self.sigma / normK ** 2
+            
+        if x_init is None:
+            self.x = self.operator.domain_geometry().allocate()
+        else:
+            self.x = x_init.copy()
+         
+        # allocate space for operator direct & adjoint    
+        self.tmp_dir = self.operator.range_geometry().allocate()
+        self.tmp_adj = self.operator.domain_geometry().allocate()            
+            
+        self.z = self.operator.range_geometry().allocate() 
+        self.u = self.operator.range_geometry().allocate() 
+
+        self.update_objective()
+        self.configured = True  
+        
+        print("{} configured".format(self.__class__.__name__, ))
+        
+    def update(self):
```

Comment:
Does the printing allow a verbose printing including primal, dual and gap like PDHG?